src/libraries/Native/Unix/System.Net.Security.Native/extra_libs.cmake

```diff
@@ -0,0 +1,23 @@
+
+function(append_extra_security_libs NativeLibsExtra)
+  if (HAVE_GSSFW_HEADERS)
+     find_library(LIBGSS NAMES GSS)
+     if(LIBGSS STREQUAL LIBGSS-NOTFOUND)
+        message(FATAL_ERROR "Cannot find GSS.Framework and System.Net.Security.Native cannot build without it. Try installing GSS.Framework (or the appropriate package for your platform)")
+     endif()
+  elseif(HAVE_HEIMDAL_HEADERS)
+     find_library(LIBGSS NAMES gssapi)
+     if(LIBGSS STREQUAL LIBGSS-NOTFOUND)
+        message(FATAL_ERROR "Cannot find libgssapi and System.Net.Security.Native cannot build without it. Try installing heimdal (or the appropriate package for your platform)")
+     endif()
+  elseif(HeimdalGssApi)
+       message(FATAL_ERROR "HeimdalGssApi option was set but gssapi headers could not be found and System.Net.Security.Native cannot build without the headers. Try installing heimdal (or the appropriate package for your platform)")
+  else()
+     find_library(LIBGSS NAMES gssapi_krb5)
+     if(LIBGSS STREQUAL LIBGSS-NOTFOUND)
+        message(FATAL_ERROR "Cannot find libgssapi_krb5 and System.Net.Security.Native cannot build without it. Try installing libkrb5-dev (or the appropriate package for your platform)")
+     endif()
+  endif()
+
+  set(${NativeLibsExtra} ${${NativeLibsExtra}} ${LIBGSS} PARENT_SCOPE)
+endfunction(append_extra_compression_libs)
```

Comment:
This is causing a new warning. cmake is expecting `endfunction()` or matching arguments: `endfunction(append_extra_compression_libs NativeLibsExtra)`.

Reply:
> new warning

There is an existing warning about cmake_policy that shows up in Linux and Android libraries builds. PR's builds are showing it up in CLR builds too because of libs linking. The fix is pretty trivial though:

```diff
--- a/src/libraries/Native/Unix/configure.cmake
+++ b/src/libraries/Native/Unix/configure.cmake
@@ -7,6 +7,11 @@ include(CheckStructHasMember)
 include(CheckSymbolExists)
 include(CheckTypeSize)
 
+# CMP0075 Include file check macros honor CMAKE_REQUIRED_LIBRARIES.
+if(POLICY CMP0075)
+    cmake_policy(SET CMP0075 NEW)
+endif()
+
 if (CLR_CMAKE_TARGET_ANDROID)
     set(PAL_UNIX_NAME \"ANDROID\")
 elseif (CLR_CMAKE_TARGET_BROWSER)
```
then we are warning free, as far as cmake is concerned. :)

Reply:
it was just copy/paste error. Fixed it while changing functions to macros.

Reply:
Both warnings disappeared on tip of you branch. :)
On illumos, I found a regression from master vs. libQ branch. This tiny patch fixes the build: https://github.com/am11/runtime/commit/a79c37e0009748336d9eef58b6eb65a962711f2c, I tested globalization APIs etc. and they are working as before. Could you apply it? I can also follow up later.

```sh
$ RFS: ROOTFS_DIR=/crossrootfs/x64
$ DOTNET_DOCKER_TAG: mcr.microsoft.com/dotnet-buildtools/prereqs:ubuntu-18.04-cross-illumos-20201123165548-f13d79e
$ docker run -e $RFS -v $(pwd):/runtime $DOTNET_DOCKER_TAG /runtime/build.sh -c Release -cross -os illumos -gcc
```

Reply:
I have added the change.

Reply:
Thanks, looks like clang is not happy with it, we can change it to:

```diff
        case PAL_RLIMIT_NOFILE:
            return RLIMIT_NOFILE;
#if !defined(PAL_RLIMIT_RSS) || !defined(PAL_RLIMIT_NPROC)
        default:
            break;
#endif
    }
```

since the error was ([logs](https://github.com/am11/runtime/runs/1495997233?check_suite_focus=true)):
```
  /runtime/src/libraries/Native/Unix/System.Native/pal_process.c:534:5: error: enumeration value 'PAL_RLIMIT_RSS' not handled in switch [-Werror=switch]
       switch (value)
       ^~~~~~
  /runtime/src/libraries/Native/Unix/System.Native/pal_process.c:534:5: error: enumeration value 'PAL_RLIMIT_NPROC' not handled in switch [-Werror=switch]
  /runtime/src/libraries/Native/Unix/System.Native/pal_process.c: At top level:
```

Reply:
@am11 hmm. This needs more thought. 
On some OS-es the switch covers all possible enum values and adding default does not compile. There is probably another way to fix, but we can do this after this merges.

Reply:
We posted at the same time and missed each others comments. 🤣
Sure, I'll follow up. Thanks for this, among other coolness it is bringing, I have noticed 0.3 MB size reduction in final .tar.gz (under artifacts/packages/{Config}/Shipping). 🚀